ProcessLib/TES/TESLocalAssemblerInner-impl.h

```diff
@@ -0,0 +1,372 @@
+{
+    switch (var)
+    {
+        case TESIntPtVariables::REACTION_RATE:
+            return _d.reaction_rate;
+        case TESIntPtVariables::SOLID_DENSITY:
+            return _d.solid_density;
+        case TESIntPtVariables::VELOCITY_X:
+            return _d.velocity[0];
+        case TESIntPtVariables::VELOCITY_Y:
+            assert(_d.velocity.size() >= 2);
+            return _d.velocity[1];
+        case TESIntPtVariables::VELOCITY_Z:
+            assert(_d.velocity.size() >= 3);
+            return _d.velocity[2];
+
+        case TESIntPtVariables::LOADING:
+        {
+            auto& Cs = cache;
+            Cs.clear();
+            Cs.reserve(_d.solid_density.size());
+
+            for (auto rho_SR : _d.solid_density)
+            {
+                Cs.push_back(rho_SR / _d.ap.rho_SR_dry - 1.0);
+            }
+
+            return Cs;
+        }
+
+        // TODO that's an element value, ain't it?
+        case TESIntPtVariables::REACTION_DAMPING_FACTOR:
+        {
+            auto const num_integration_points = _d.solid_density.size();
+            auto& alphas = cache;
+            alphas.clear();
+            alphas.resize(num_integration_points,
+                          _d.reaction_adaptor->getReactionDampingFactor());
+
+            return alphas;
+        }
+    }
+
+    cache.clear();
+    return cache;
+}
+
+template <typename Traits>
+void TESLocalAssemblerInner<Traits>::assembleIntegrationPoint(
+    unsigned integration_point,
+    std::vector<double> const& localX,
+    const typename Traits::ShapeMatrices::ShapeType& smN,
+    const typename Traits::ShapeMatrices::DxShapeType& smDNdx,
+    const typename Traits::ShapeMatrices::JacobianType& smJ,
+    const double smDetJ,
+    const double weight,
+    typename Traits::LocalMatrix& local_M,
+    typename Traits::LocalMatrix& local_K,
+    typename Traits::LocalVector& local_b)
+{
+    preEachAssembleIntegrationPoint(integration_point, localX, smN, smDNdx, smJ,
+                                    smDetJ);
+
+    auto const N = smDNdx.cols();  // number of integration points
+    auto const D = smDNdx.rows();  // global dimension: 1, 2 or 3
+
+    assert(N * NODAL_DOF == local_M.cols());
+
+    auto const laplaceCoeffMat = getLaplaceCoeffMatrix(integration_point, D);
+    assert(laplaceCoeffMat.cols() == D * NODAL_DOF);
+    auto const massCoeffMat = getMassCoeffMatrix(integration_point);
+    auto const advCoeffMat = getAdvectionCoeffMatrix(integration_point);
+    auto const contentCoeffMat = getContentCoeffMatrix(integration_point);
+
+    // calculate velocity
+    assert((unsigned)smDNdx.rows() == _d.velocity.size() &&
+           (unsigned)smDNdx.cols() == _d.velocity[0].size());
+
+    auto const velocity =
+        (Traits::blockDimDim(laplaceCoeffMat, 0, 0, D, D) *
+         (smDNdx * Eigen::Map<const typename Traits::Vector1Comp>(localX.data(),
+                                                                  N)  // grad_p
+          /
```

Comment:
It's automatically formatted by clang-format and difficult to get it consistently formatted into an 80 char bound simply because the `Eigen::Map...(..,N)` subexpression is very long.